server/applicationPattern/applicationPattern/onfModel/services/ForwardingConstructAutomationServices.test.js

```diff
@@ -0,0 +1,439 @@
+    const operationServerName = '/v1/embed-yourself';
+    
+    jest.spyOn(ControlConstruct, 'getLogicalTerminationPointListAsync').mockImplementation(() => ltpsForSubscription);
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => undefined);    
+    const res =  ForwardingConstructAutomationServices.automateForwardingConstructAsync(
+      operationServerName, forwardingAutomationInputListforSubscription, 'User Name', 
+      '550e8400-e29b-11d4-a716-446655440000', '1.3.1', 'Unknown value');
+    return res.then()
+    .catch((err)=> {
+      expect(err).toBeInstanceOf(Error);
+    })
+  });
+
+  test("automateForwardingConstructAsync - failed to automate ForwardingConstructAsync when FC Port is subscription", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    const mockError = { message: 'Something bad happened' };
+    
+    jest.spyOn(ControlConstruct, 'getLogicalTerminationPointListAsync').mockImplementation(() => ltpsForSubscription);
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fcForSubscription);
+    jest.spyOn(event, 'dispatchEvent').mockImplementation(() => true);
+     
+    const res =  ForwardingConstructAutomationServices.automateForwardingConstructAsync(
+      operationServerName, forwardingAutomationInputListforSubscription, 'User Name', 
+      '550e8400-e29b-11d4-a716-446655440000', '1.3.1', 'Unknown value');
+    return res.then(() => {
+      return Promise.reject(mockError);
+    })
+    .catch((err)=> {
+      expect(err).toEqual(mockError);
+    })
+  });
+
+})
+
+describe("automateForwardingConstructWithoutInputAsync", () => {
+  test("automateForwardingConstructWithoutInputAsync - failed to get ForwardingConstructForTheForwardingNameAsync", async () => {
+    const mockError = { message: 'Something bad happened' };
+
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => {
+      return Promise.reject(mockError);
+    });
+
+    const res = ForwardingConstructAutomationServices.automateForwardingConstructWithoutInputAsync(
+        forwardingAutomationInputList, 'User Name', '550e8400-e29b-11d4-a716-446655440000',
+        '1.3.1', 'Unknown value');
+
+    return res.then()
+      .catch((err)=> {
+        expect(err).toEqual(mockError);
+    })
+  });
+
+  test("automateForwardingConstructWithoutInputAsync - if ForwardingConstructForTheForwardingNameAsync returns undefined", async () => {
+
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => undefined);
+
+    const res = ForwardingConstructAutomationServices.automateForwardingConstructWithoutInputAsync(
+        forwardingAutomationInputList, 'User Name', '550e8400-e29b-11d4-a716-446655440000',
+        '1.3.1', 'Unknown value');
+
+    return res.then()
+      .catch((err)=> {
```

Comment:
This test never reaches the catch block. As I wrote before, if you put expect in catch block and the catch block is not reached, test will pass without ever reaching the expect.